File: server/java-api/myproject/src/main/java/com/javaapi/app/user/usecase/Session/SessionUsecase.java
package com.javaapi.app.user.usecase.Session;

import org.springframework.stereotype.Service;

import com.javaapi.app.user.core.domain.model.vo.Email;
import com.javaapi.app.user.core.domain.model.vo.Userid;
import com.javaapi.app.user.core.domain.model.vo.Username;
import com.javaapi.app.user.core.dto.SessionDTO;
import com.javaapi.app.user.framework.session.SessionStore;


@Service
public class SessionUsecase{

    private final SessionFactory sessionFactory;
    private final SessionStore sessionStore;

    public SessionUsecase(SessionFactory sessionFactory,SessionStore sessionStore) {
        this.sessionFactory = sessionFactory;
        this.sessionStore = sessionStore;
    }


    public String createUserSession(SessionDTO sessionDTO,String session){
        sessionFactory.createUserSession(sessionDTO,session);

        return "OK";
    }

    public SessionDTO getUserSession(String session){

        SessionDTO userInfo = sessionStore.getUserInfo(session);

        System.out.println("🐞 Retrieved UserInfo: userId=" + userInfo.getUserId());
        Userid valUserid = new Userid(userInfo.getUserId());
        Username valUsername = new Username(userInfo.getUsername());
        Email valEmail = new Email(userInfo.getEmail());

        SessionDTO sessionDTO = new SessionDTO(valUserid.getUserid(), valUsername.getUsername(), valEmail.getEmail());
        return sessionDTO;
    }
}

File: server/go-ws/internal/app/usecase/session/SessionUsecase.go
package session

import (
	"context"
	dto "go-ws/internal/app/core/dto/session"
	"go-ws/internal/app/framnework/session"
)

type SessionUsecase struct {
	sessionStore *session.SessionStore
	sessionFactory *SessionFactory
}

func NewSessionUsecase(sessionStore *session.SessionStore) *SessionUsecase {
	return &SessionUsecase{sessionStore: sessionStore, sessionFactory: NewSessionFactory()}
}

func (u *SessionUsecase) GetSession(ctx context.Context, sessionID string) (dto.SessionDTO, error) {
	session, err := u.sessionStore.GetSession(ctx, sessionID)
	if err != nil {
		return dto.SessionDTO{}, err
	}

	sessiondto := u.sessionFactory.CreateSession(session)
	return *sessiondto, nil

}